Better logging and exceptions

Ercadio opened the pull request.

Ercadio commented on src/util/Exception.cxx:
```diff
@@ -0,0 +1,63 @@
+
+#ifdef _WIN32
+/**
+ *  @todo implement window stack trace
```
Add SIG_handler for Win32

bangboomboomboom commented on src/engine/UIManager.cxx:
```diff
@@ -288,30 +282,16 @@ void UIManager::drawUI() const
-    // cast the object to a Button to check if it's a toggle button.
-    if (button && button->getUiElementData().isToggleButton)
-    {
-      for (const auto &it : m_uiGroups[groupID])
```
I'm not sure why this one statement has been removed, it doesn't really seem to be part of this branch. In anycase, this loop sets other buttons in this group to have the same visibility as the button that was pressed. Like this, all the buttons are either visible or invisible, rather than just this group. The way this works isn't great but I think that we'll need to keep this loop in here for now, this entire class will have to be refactored in the future and hopefully that makes it a bit more straight forward.

Ercadio replied:
Hmm you're right. I thought we had duplicated code but I was wrong

adct-the-experimenter commented:
Unfortunately, I get this error when trying to start the game after building it.

terminate called after throwing an instance of 'std::__ios_failure'
  what():  basic_ios::clear: iostream error

I built it in Arch Linux with the cmake options
cmake -DENABLE_SDL2_MIXER=1 -DENABLE_OPENAL_SOFT=1 -DUSE_PACKAGE_MANAGER=0 -DBUILD_TEST=0 -DENABLE_MOFILEREADER=0 --build ..

I made sure to run make copy_resources as well.

Ercadio commented:
> Unfortunately, I get this error when trying to start the game after building it.
> 
> terminate called after throwing an instance of 'std::__ios_failure'
> what(): basic_ios::clear: iostream error
> 
> I built it in Arch Linux with the cmake options
> cmake -DENABLE_SDL2_MIXER=1 -DENABLE_OPENAL_SOFT=1 -DUSE_PACKAGE_MANAGER=0 -DBUILD_TEST=0 -DENABLE_MOFILEREADER=0 --build ..
> 
> I made sure to run make copy_resources as well.

I was just able to fix this. I moved this out of WIP because I also added & tested the SIG Handler on Windows

Ercadio commented:
I will close & reopen this PR
